activesupport/lib/active_support/number_helper/number_to_currency_converter.rb

```diff
@@ -11,9 +11,16 @@ def convert
         number = self.number.to_s.strip
         format = options[:format]
 
-        if number.sub!(/^-/, "") &&
-           (options[:precision] != 0 || number.to_f > 0.5)
-          format = options[:negative_format]
+        if number.sub!(/^-/, "")
+          number_f = number.to_f.abs
+          if number_f == 0.0
+            # likely an alternate input format that failed to parse.
+            # see https://github.com/rails/rails/pull/39350
+            format = options[:negative_format]
+          else
+            number_f *= 10**options[:precision]
```

Comment:
```suggestion
            number_f *= 10 ** options[:precision]
```

Reply:
I originally had it with spaces and rubocop failed it saying the spaces needed to be removed.

Reply:
```
lib/active_support/number_helper/number_to_currency_converter.rb:21:28: C: [Correctable] Layout/SpaceAroundOperators: Space around operator ** detected.
            number_f *= 10 ** options[:precision]
                           ^^
```

Reply:
What a strange cop, the idea was to avoid ` number_f*=` but it also changing the `**`. That is fine then.